ppts/views.py

```diff
@@ -1,6 +1,77 @@
 from django.shortcuts import render
 from django.http import HttpResponse
+from django.http import Http404
+
+import matplotlib.pyplot as plt
+import PIL, PIL.Image
+import io
+from matplotlib.backends.backend_agg import FigureCanvasAgg as FigureCanvas
+
+from ppts.models import DwellingType
+from ppts.models import LandUse
+from ppts.models import Location
+from ppts.models import Planner
+from ppts.models import ProjectFeature
+from ppts.models import ProjectDescription
+from ppts.models import Record
+from ppts.models import RecordType
+from django.db.models import Count
 
 def index(request):
-    context = {}
-    return render(request, 'ppts/index.html', context)
\ No newline at end of file
+    graph_list = []
+    
+    #append names of graphs
+    graph_list.append('samplegraph')
+    graph_list.append('samplegraph2')
+    graph_list.append('samplegraph')
+    
+    #use list of graphs as context and render page
+    context = {'graph_list': graph_list}
+    return render(request, 'ppts/index.html', context)
+
+def graphmanager(request, graphname):
+    '''given the name of the graph, sends request to appropriate function'''
+    graph_func = globals()['graph_' + graphname]
+    return figureToHttp(graph_func())
+    #to be added: 404 error when there's no graph
+
+def graph_samplegraph():
+    # Construct the graph
+    fig,ax = plt.subplots()
+    x = range(0, 20, 1)
+    s = range(0,40,2)
```

Comment:
nit: There are style inconsistencies in the file. That's fine, but most companies and organizations enforce some kind of style check to ensure the code sticks to a consistent style so that every developer can interact with and understand each others' code. We don't do this yet, but it's likely something we'll do in the future.

Consider running flake8 or pylint on your source code, or adding it as an automatic hook in your code editor. For example, I use vim and I have a setting that runs flake8 on the file every time I hit save. That way I have to make the code conform to a consistent style unless I want to be continuously bugged about style errors.